### PR TITLE
Add tests for message encoding and decoding

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

**Comment by adutra** [comment]
TODO

- EVENT
- RESULT

**Comment by adutra** [comment]
@alicel and @joao-r-reis I added tests for the missing messages (ERROR, EVENT, RESULT). I confess I did not cover all the possible error cases as the amount of test code was already huge. IMO we can add these later but I wouldn't want this PR to stay open for too long because it would block further development.

File: cassandraprotocol/constants.go [hunk 1]
```diff
@@ -260,3 +208,15 @@ const (
 	RowsFlagNoMetadata       = RowsFlag(0x00000004)
 	RowsFlagMetadataChanged  = RowsFlag(0x00000008)
 )
+
+type VariablesFlag = int32
+
+const (
+	VariablesFlagGlobalTablesSpec = VariablesFlag(0x00000001)
```
Comment (joao-r-reis) [inline cassandraprotocol/constants.go:215]: I tried to figure out what this is but I couldn't, can you give me a brief explanation?

**Review by joao-r-reis — APPROVED** [review]
There's a ton of work here, awesome job! I added a comment (it's just a question)

File: cassandraprotocol/constants.go (hunk 1 above)
Comment (adutra) [reply cassandraprotocol/constants.go:215]: It's a reference to the sole flag defined for `PreparedResult` bound variables metadata, see [here](https://github.com/datastax/go-cassandra-native-protocol/blob/main/specs/native_protocol_v4.spec#L653).

For a bit of context: the Java equivalent of this project defines one single object called `RowsMetadata` and applies it to both `PreparedResult` and `RowsResult` messages in 3 places:

1. To model the `PreparedResult` `<metadata>` structure
2. To model the `PreparedResult` `<result_metadata>` structure
3. To model the `RowsResult` `<metadata>` structure

Indeed, 2 and 3 are identical (barring the fact that not all flags are meaningful depending on the message), but in fact, 1 is not 100% identical to 2 and 3. 

After struggling with this in Go for a while, I decided that it was easier to create a separate struct for 1 above, which is why in this PR you see a `RowsMetadata` (for 2 and 3 above) and also a new `VariablesMetadata` struct (for 1 above). For the same reason, I preferred to create a separate `VariablesFlag` that represents the flags of a `VariablesMetadata` struct.

Comment (joao-r-reis) [reply cassandraprotocol/constants.go:215]: Thanks for the explanation! I've never looked too much into the prepared metadata so this is very useful 👍

**Pull request merged**